benchmark/parse_cdata.yaml

```diff
@@ -0,0 +1,50 @@
+loop_count: 100
+contexts:
+  - gems:
+      rexml: 3.2.6
+    require: false
+    prelude: require 'rexml'
+  - name: master
+    prelude: |
+      $LOAD_PATH.unshift(File.expand_path("lib"))
+      require 'rexml'
+  - name: 3.2.6(YJIT)
+    gems:
+      rexml: 3.2.6
+    require: false
+    prelude: |
+      require 'rexml'
+      RubyVM::YJIT.enable
+  - name: master(YJIT)
+    prelude: |
+      $LOAD_PATH.unshift(File.expand_path("lib"))
+      require 'rexml'
+      RubyVM::YJIT.enable
+
+prelude: |
+  require 'rexml/document'
+  require 'rexml/parsers/sax2parser'
+  require 'rexml/parsers/pullparser'
+  require 'rexml/parsers/streamparser'
+  require 'rexml/streamlistener'
+
+  def build_xml(n_depth)
+    xml = "<?xml version=\"1.0\"?>\n" +
+           "<root>Test</root>\n" +
+           "<!" + "[CDATA[" * n_depth + "]]>\n"
```

Comment:
I think `"<![CDATA[" + "a" * size + "]]>"` is better because it is clear that this benchmark is testing a large CDATA, not nested CDATA.
As commented in https://github.com/ruby/rexml/issues/243#issuecomment-2692007665, NESTED is not important.
(And since this is not a performance bug fix, perhaps benchmarking is not so important)

Reply:
> (And since this is not a performance bug fix, perhaps benchmarking is not so important)

Right.

If it's a performance bug, we should add a test that uses `assert_linear_performance` or something. But this is not a performance bug, this is just an optimization. So benchmark is suitable than test.

Reply:
Yes.
This PR is an improvement, not a bug, so I'll remove the `GitHub: fix` comment.

Reply:
Ah, `GitHub: fix` is needed for auto issue close on merge. Please keep it.

Reply:
> Please keep it.

OK